crates/tuono/src/cli.rs

```diff
@@ -47,15 +47,58 @@ fn init_tuono_folder(mode: Mode) -> std::io::Result<()> {
     Ok(())
 }
 
-pub fn app() -> std::io::Result<()> {
+fn extract_port(addr: &str) -> &str {
+    addr.split(":").last().unwrap_or_else(|| {
+        eprintln!("Error: Failed to extract port from address {}", addr);
+        std::process::exit(1);
+    })
+}
+
+async fn check_ports(mode: Mode) -> std::io::Result<()> {
+    println!("\nChecking ports...");
+
+    let rust_addr = "0.0.0.0:3000";
+    let rust_port = extract_port(rust_addr);
```

Comment:
I'd create two static constants 
```
const TUONO_PORT: u16 = 3000;
const VITE_PORT: u16 = 3001;
```
And then just bind the listener with `std::net::TcpListener::bind(format!("0.0.0.0:{TUONO_PORT}"))`

I don't think we need to extract it from the static `0.0.0.0:3000` address